### DISCUSSION TITLE
[Feature] Tiered Media Storage for ZFS

### DISCUSSION THREAD

**Discussion by AlexeyMatskevich** [immich-app/immich#14907] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello! Thank you for your excellent software.
I am using Immich to store my personal photos and videos. In my case, I have many photos taken on smartphones and mirrorless cameras, with an average size ranging from 1 MB to 10 MB. I also have plenty of videos with durations from 10 minutes to 2 hours, ranging from 2 GB to 60 GB in size. Additionally, I use the OpenZFS file system on my Immich server.

My current issue is related to the Recordsize parameter in OpenZFS. By default, it’s set to 128 KB, which is well-suited for storing photos and other small files. However, for longer videos, an optimal block size could be 1 MB. Since Immich places all photos and videos in the same folder, I cannot separate them into different folders to specify distinct OpenZFS settings for each type of content.

My request is to enable Immich to store photos and videos in different locations or, for instance, to place files in different storage locations based on their size (for example, files up to 50 MB in one location, and files larger than 50 MB in another).

In addition, by naming several "folders" I could set different compression parameters and other parameters, which is convenient considering that videos are often incompressible, unlike photographs.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
Dupe of #4620. You can separate files by type on disk using the storage template.

**Reply by AlexeyMatskevich** [reply]
All the same it does not look like the same thing, in the above request want the ability to mix local storage and remote like s3, I just need to be able to specify the path to directories separately by content type or file size.
